src/main/java/com/internship/backend/controller/ReservationController.java

```diff
@@ -32,24 +29,25 @@ public class ReservationController {
     private ReservationService reservationService;
 
     @PostMapping("/add")
-    public ResponseEntity<Reservation> addReservation(@RequestBody ReservationDTO reservationDTO) {
+    public ResponseEntity<Reservation> addReservation(@Valid @RequestBody ReservationDTO reservationDTO) {
```

Comment:
Could you explain the purpose of adding @Valid on the method? What is expected to be validated? And which fields of the model should be validated?

Reply:
The @Valid annotation triggers validation of the ReservationDTO object based on validation constraints defined in its fields (in Reservation class I am using @NotNull annotation). It prevents null data from being processed further (for startTime and endTime fields)
I think I should have used the @NotNull in ReservationDTO, not in Reservation class.